desc/io/ascii_io.py

```diff
@@ -14,17 +24,51 @@ def write_ascii(fname, eq):
         dictionary of equilibrium parameters.
 
     """
+    warnings.warn(
+        "text-based IO is not as well-tested or maintained as the .h5"
+        " format and is therefore not recommended, please use the"
+        " .h5 output format instead. text-based IO will be deprecated"
+        " in a future release",
+        DeprecationWarning,
+    )
+    eq = eq.copy()  # so don't mess up original one
+    if eq.iota is None:
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore")
+            eq.iota = eq.get_profile("iota", kind="power_series")
+    from desc.profiles import FourierZernikeProfile, PowerSeriesProfile
+
     # can't import other stuff in io due to circular imports, so have to check by name
-    assert (
-        eq.pressure.__class__.__name__ == "PowerSeriesProfile"
-        and eq.iota.__class__.__name__ == "PowerSeriesProfile"
-    ), "Equilibrium must have power series profiles for ascii io"
+    errorif(
+        isinstance(eq.pressure, FourierZernikeProfile)
+        or isinstance(eq.iota, FourierZernikeProfile),
+        ValueError,
+        "cannot save FourierZernikeProfile as ascii",
+    )
 
     pressure = eq.pressure
     iota = eq.iota
+
+    if not isinstance(iota, PowerSeriesProfile):
+        iota = iota.to_powerseries(order=eq.L, sym=True)
+    if not isinstance(pressure, PowerSeriesProfile):
+        pressure = pressure.to_powerseries(order=eq.L, sym=True)
```

Comment:
if the equilibrium has kinetic profiles then `pressure` is `None` so I think this will fail, but it should always be safe to do `eq.get_profile("pressure", kind="power_series")`